packages/auth-core/src/core.ts

```diff
@@ -0,0 +1,277 @@
+import jwtDecode from "jwt-decode";
+import * as edgedb from "edgedb";
+import { ResolvedConnectConfig } from "edgedb/dist/conUtils";
+
+import * as pkce from "./pkce";
+import { BuiltinOAuthProviderNames, emailPasswordProviderName } from "./consts";
+
+export interface TokenData {
+  auth_token: string;
+  identity_id: string | null;
+  provider_token: string | null;
+  provider_refresh_token: string | null;
+}
+
+export class Auth {
+  /** @internal */
+  public readonly baseUrl: string;
+
+  protected constructor(
+    public readonly client: edgedb.Client,
+    baseUrl: string
+  ) {
+    this.baseUrl = baseUrl;
+  }
+
+  static async create(client: edgedb.Client) {
+    const connectConfig: ResolvedConnectConfig = (
+      await (client as any).pool._getNormalizedConnectConfig()
+    ).connectionParams;
+
+    const [host, port] = connectConfig.address;
+    const baseUrl = `${
+      connectConfig.tlsSecurity === "insecure" ? "http" : "https"
+    }://${host}:${port}/db/${connectConfig.database}/ext/auth`;
```

Comment:
@jaclarke  There is missing slash `/`

```js
new URL("foo", "http://localhost:3000/auth").toString()
// 'http://localhost:3000/foo'
new URL("foo", "http://localhost:3000/auth/").toString()
// 'http://localhost:3000/auth/foo'
```

This causes wrong builtin urls to be created (beside other)

![image](https://github.com/edgedb/edgedb-js/assets/38855190/9694d55b-0e66-42e2-98e5-8d3b9aba2225)

Reply:
Yep, you're right. Thanks for catching that.